packages/wdio-browserstack-service/src/service.ts

```diff
@@ -1,12 +1,72 @@
 import logger from '@wdio/logger'
-import got from 'got'
+import got, { GotRequestFunction } from 'got'
+import WebDriver from 'webdriver'
 
 import { getBrowserDescription, getBrowserCapabilities, isBrowserstackCapability } from './util'
 
 const log = logger('@wdio/browserstack-service')
 
+type Capabilities = WebDriver.Capabilities & WebdriverIO.MultiRemoteCapabilities;
+
+type Browser = WebdriverIO.BrowserObject & WebdriverIO.MultiRemoteBrowserObject;
+
+type Config = WebdriverIO.Config & { cucumberOpts?: { strict: boolean } };
+
+type Suite = {
+    title: string;
+}
+
+type Test = {
+    fullName: string,
+    parent: string,
+    title: string
+};
+
+type Context = any;
+
+type Specs = any;
+
+type Feature = {
+    document: {
+        feature: {
+            name: string;
+        }
+    }
+};
+
+type Results = {
+    error: Error;
+    passed: boolean
+    exception: any;
+    status: string;
+};
+
+type Pickle = {
+    name: string;
+}
+
+type MultiRemoteAction = (sessionId: string, browserName?: string) => ReturnType<GotRequestFunction> | Promise<void>;
+
+type SessionResponse = {
+    // eslint-disable-next-line camelcase
+    automation_session: {
+        // eslint-disable-next-line camelcase
+        browser_url: string
+    }
+}
+
 export default class BrowserstackService {
-    constructor (options = {}, caps, config) {
+    config: Config;
+    sessionBaseUrl: string;
+    failReasons: string[];
+    scenariosThatRan: string[];
+    preferScenarioName: boolean;
+    strict: boolean;
+    failureStatuses: string[];
+    caps: Capabilities;
+    browser?: Browser;
+    fullTitle?: string;
```

Comment:
Can we make them private starting with an `_`?

```suggestion
    private _config: Config;
    private _sessionBaseUrl: string;
    private _failReasons: string[];
    private _scenariosThatRan: string[];
    private _preferScenarioName: boolean;
    private _strict: boolean;
    private _failureStatuses: string[];
    private _caps: Capabilities;
    private _browser?: Browser;
    private _fullTitle?: string;
```

Reply:
Done and fixed all references throughout the file!